public/core.d.ts

```diff
@@ -4596,6 +4596,94 @@ declare namespace JMap {
        */
       function removeByIds(basemapIds: string[]): void
     }
+
+    namespace Attribution {
+      /**
+       * ***JMap.Map.Attribution.getAll***
+       * 
+       * Returns all the attributions displayed on the map.
+       * 
+       * @example ```ts
+       * 
+       * // Get all the attributions currently displayed.
+       * const attributions: JMapAttribution = JMap.Map.Attribution.getAll()
+       * 
+       * ```
+       */
+      function getAll(): JMapAttribution[]
+
+      /**
+       * ***JMap.Map.Attribution.addSingle***
+       * 
+       * Add a custom attribution on the map.
+       * 
+       * @throws Errors if incorrect or taken id, if both text and imgSrc are not a string, if href is initialized but not a string
+       * @param attribution the attribution to add to the map
+       * @example ```ts
+       *
+       * // Add a link attribution on the map.
+       *
+       * JMap.Map.Attribution.addSingle({ id: "link-test", text: "© HelloWorld", href:"https://k2geospatial.com/jmap-en/"})
+       * 
+       * // Add a text attribution on the map.
+       *
+       * JMap.Map.Attribution.addSingle({ id: "text-test", text: "© HelloWorld")
+       *
+       * // Add a image attribution on the map.
+       *
+       * JMap.Map.Attribution.addSingle({id: "test-image",imgSrc: "https://k2geospatial.com/wp-content/themes/k2-theme/assets/images/k2-logo.png", href: "https://k2geospatial.com/jmap-en/"})
+       * ```
+       */
+      function addSingle(attribution: JMapAttribution): void
+
+      /**
+       * ***JMap.Map.Attribution.addMultiple***
+       * 
+       * Add a custom attribution on the map.
+       * 
+       * @throws Errors if attributions is not an array, if one of the attribution is invalid
```

Comment:
Errors if invalid parameters, or if an attribution having the same id already exists